ledger-tool/src/ledger_utils.rs

```diff
@@ -54,6 +54,19 @@ use {
     thiserror::Error,
 };
 
+pub(crate) struct LoadAndProcessLedgerOutput {
+    pub bank_forks: Arc<RwLock<BankForks>>,
+    pub starting_snapshot_hashes: Option<StartingSnapshotHashes>,
+    // Typically, we would want to join all threads before returning. However,
+    // AccountsBackgroundService (ABS) performs several long running operations
+    // that don't respond to the exit flag. Blocking on these operations could
+    // significantly delay getting results that do not need ABS to finish. So,
+    // skip joining ABS and instead let the caller decide whether to block or
+    // not. It is safe to let ABS continue in the background, and ABS will stop
+    // if/when it finally checks the exit flag
+    pub accounts_background_service: AccountsBackgroundService,
+}
```

Comment:
> nit: Why have the type itself use a different visibility that's less visible? Sometimes I see different visibility, but it's with the type having more visibility.

`pub(crate)` on the type means nobody outside of the crate can utilize the type. With that noted, `pub` and `pub(crate)` on individual fields are essentially equivalent. Even though the fields are `pub`, some other dependent crate still can't import the type.

> Since nothing depends on ledger-tool, seems fine to make this type pub. Wdyt?

I don't feel too strongly. Hypothetically, I'd argue that everything in `ledger-tool` should be `pub(crate)` given that we change functions / types / etc whenever. However, I also don't see any dependents outside of our tree either:
https://crates.io/crates/agave-ledger-tool/reverse_dependencies

Given that so much other stuff is `pub`, I could see an argument for making this `pub` as well for the sake of consistency

Reply:
I pushed my other PR first which conflicted with this one, so decided to take another look.

> Can binaries even export types and/or having other libs/bins import them?

Good point - I think you're right that nothing else can import a type from `ledger-tool` as-is. I believe my comment is about `pub(crate)` on a type "over-ruling" `pub` on its' fields is valid for lib crates. Maybe you were speaking about ledger-tool specifically (as a bin, not a leaf-node crate) and I misinterpreted you 😅.

In any case, since I'm about to rebase in order to resolve the merge-conflict, I'll adjust `pub(crate)` to `pub`. As mentioned, I'm good with this as it is consistent with the rest of the module too.

Change is in a new commit - force push was to rebase in order to solve merge conflict